internal/cluster/cluster_test.go

```diff
@@ -2836,24 +2841,26 @@ var _ = Describe("Deregister inactive clusters", func() {
 	})
 
 	It("Deregister inactive cluster limited", func() {
-		inactiveCluster1 := registerCluster()
-		inactiveCluster2 := registerCluster()
-		inactiveCluster3 := registerCluster()
-		inactiveCluster4 := registerCluster()
-		inactiveCluster5 := registerCluster()
-		inactiveCluster6 := registerCluster()
+		var (
+			clustersBeforeDeregisterInactive []*common.Cluster
+			clustersAfterDeregisterInactive  []*common.Cluster
+		)
+
+		for i := 0; i < 6; i++ {
+			registerCluster()
+		}
 
 		lastActive := strfmt.DateTime(time.Now())
 
-		Expect(state.DeregisterInactiveCluster(ctx, 3, lastActive)).ShouldNot(HaveOccurred())
+		err := db.Find(&clustersBeforeDeregisterInactive).Error
```

Comment:
minor comment but you can use count instead of find.

Reply:
I didn't find other places in the code where we do that